server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -468,8 +469,13 @@ public Pool find(String poolId) {
             entitleDate, owner, null, fromPools);
 
         if (bestPools == null) {
-            throw new RuntimeException("No entitlements for products: " +
-                Arrays.toString(productIds));
+            List<String> fullList = new ArrayList<String>();
+            fullList.addAll(Arrays.asList(productIds));
+            for (ConsumerInstalledProduct cip : consumer.getInstalledProducts()) {
+                fullList.add(cip.getId());
+            }
+            log.warn("No entitlements available for products: " + fullList);
```

Comment:
Why did the consumer installed products come into play here? I think the list of productIds is already set to what was requested, which would normally be their installed products.